nfs/inc/rpc_task.h

```diff
@@ -437,6 +437,37 @@ struct rpc_task
 
     // Put a cap on how many async tasks we can start.
     static std::atomic<int> async_slots;
+
+public:    
+    /*
+     * Total number of parallel reads issued to backend.
+     * A single RPC read task can result in issuing multiple read calls to the
+     * backend server to decrease the processing time, hence this wlll be used
+     * to keep track of these issued reads.
+     * This is valid only for reads.
+     * This should be updated after taking the lock read_task_lock.
+     */
+    int num_of_reads_issued_to_backend;
+    
+    /*
+     * A single RPC read task can result in issuing multiple read calls to the
+     * backend server and hence we should wait till we complete all these reads
+     * before returning response to the caller.
+     * One this is set to True, the response can be safely returned to the caller.
+     * This is valid only for reads.
+     * This should be updated after taking the lock read_task_lock.
+     */
+    bool readfile_completed;
+    
+    /*
+     * This is currently valid only for reads.
+     * This contains vector of byte chunks which will be returned by making
+     * a call to bytes_chunk_cache::get().
+     */
+    std::vector<bytes_chunk> bytes_vector;
+
+    std::shared_mutex read_task_lock;
```

Comment:
we should be able to avoid the lock by making the above two atomic and carefully setting/reading them